package.json

```diff
@@ -38,6 +38,7 @@
     "url": "git+https://github.com/cocos-creator/engine.git"
   },
   "devDependencies": {
+    "@types/node": "^20.5.9",
```

Comment:
Why modify node version again?

Reply:
because there is a type issue on '@types/node' 17, see this issue
https://github.com/microsoft/TypeScript/issues/51567#issuecomment-1412901733

Reply:
this is the node type version, which should be backward compatible

it's not npm version or the node runtime environment, 
we upgrade this version only to fix type issue